compile fails on knl (Knights Landing)

When compiling with xsimd it fails on KNL architecture.
uname -a : Linux x 3.10.0-693.el7.x86_64 #1 SMP Tue Aug 22 21:09:27 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

gcc: 8.2.1 20180905 (Red Hat 8.2.1-3)
xsimd git hash: commit af9c6723b7eba63da88e42bf69baf80b601c6007
cpu: Intel(R) Xeon Phi(TM) CPU 7250 @ 1.40GHz

Example code:
`#include <xsimd/xsimd.hpp>`
`int main(int argc, char **argv) {`
`return 0;`
`}`

Compiled with 

g++ std=c++17 -march=knl tst.cpp

yields the compile error in the following log:
[err_knl.txt](https://github.com/xtensor-stack/xsimd/files/3879842/err_knl.txt)

The problem arises in

> kernel::select(cond(), a(), b());

where types are not matching (unsigned vs signed).

OK, though I dont see quite the same situation - it's in the same region of the code.
The error I see relates to mismatch in argument types. The code I tested is the latest from github.
> xsimd/types/xsimd_avx512_conversion.hpp:47:5:   required from here
> xsimd/types/../types/xsimd_base.hpp:1625:30: error: no matching function for call to 
> xsimd::detail::batch_kernel<unsigned char, 64>::select(const xsimd::batch_bool<signed char, 64>&, const xsimd::batch<unsigned char, 64>&, const xsimd::batch<unsigned char, 64>&)’
         return kernel::select(cond(), a(), b());
                ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~

--- a/include/xsimd/types/xsimd_avx512_int16.hpp
+++ b/include/xsimd/types/xsimd_avx512_int16.hpp
@@ -91,11 +91,11 @@ namespace xsimd
     };
 
     template <>
-    class batch_bool<uint16_t, 32> : public avx512_fallback_batch_bool<int16_t, 32>
+    class batch_bool<uint16_t, 32> : public avx512_fallback_batch_bool<uint16_t, 32>
     {
     public:
 
-        using base_class = avx512_fallback_batch_bool<int16_t, 32>;
+        using base_class = avx512_fallback_batch_bool<uint16_t, 32>;
         using base_class::base_class;
     };
 
@@ -110,7 +110,7 @@ namespace xsimd
 
         template <>
         struct batch_bool_kernel<uint16_t, 32>
-            : avx512_fallback_batch_bool_kernel<int16_t, 32>
+            : avx512_fallback_batch_bool_kernel<uint16_t, 32>
         {
         };
     }

--- a/include/xsimd/types/xsimd_avx512_int8.hpp
+++ b/include/xsimd/types/xsimd_avx512_int8.hpp
@@ -91,11 +91,11 @@ namespace xsimd
     };
 
     template <>
-    class batch_bool<uint8_t, 64> : public avx512_fallback_batch_bool<int8_t, 64>
+    class batch_bool<uint8_t, 64> : public avx512_fallback_batch_bool<uint8_t, 64>
     {
     public:
 
-        using base_class = avx512_fallback_batch_bool<int8_t, 64>;
+        using base_class = avx512_fallback_batch_bool<uint8_t, 64>;
         using base_class::base_class;
     };
 
@@ -110,7 +110,7 @@ namespace xsimd
 
         template <>
         struct batch_bool_kernel<uint8_t, 64>
-            : avx512_fallback_batch_bool_kernel<int8_t, 64>
+            : avx512_fallback_batch_bool_kernel<uint8_t, 64>
         {
         };
     }

